### PR TITLE
feat: add c2d env selection

### LINKED ISSUE #376: [Feature] Add selection of compute environment on C2D consume
## Motivation / Problem
We want to be able to show that a provider can make multiple compute environments accessible to the consumer. Every C2D env can be priced and configured differently.

## Solution
Add a select field to the C2D consumption step allowing to:
- [x] select a compute environment made accessible through the provider of the dataset
- [x] calculate prices and fees based on the selected enviornment
- [x] pass the selected env to the provider on start compute

## Additional context
Select box can be first step on the C2D form:
image

Prices are calculated new once a different environment gets selected
image

Request/Response for available C2D enviornments on provider:
`https://provider.v4.genx.delta-dao.com/api/services/computeEnvironments`

```json
[
    {
        "consumerAddress": "0x68C24FA5b2319C81b34f248d1f928601D2E5246B",
        "cpuNumber": 1,
        "cpuType": "",
        "currentJobs": 0,
        "desc": "",
        "diskGB": 1,
        "feeToken": "0xe974c4894996E012399dEDbda0bE7314a73BBff1",
        "gpuNumber": 0,
        "gpuType": "",
        "id": "ocean-ctd",
        "lastSeen": "1687689224.85222",
        "maxJobDuration": 600,
        "maxJobs": 100,
        "priceMin": 0,
        "ramGB": 1,
        "storageExpiry": 0
    }
]
```

for multi-chain it is something like:
```json
{
    "80001": [
      {
        "consumerAddress": "0x68C24FA5b2319C81b34f248d1f928601D2E5246B",
        "cpuNumber": 1,
        ...
      },
      {
        "consumerAddress": "0x01234",
        "cpuNumber": 2,
        ...
       }
    ],
  "100": [
    {
        "consumerAddress": "0x68C24FA5b2319C81b34f248d1f928601D2E5246B",
        "cpuNumber": 1,
        ...
      }
  ]
}
```

### PR DESCRIPTION
**Pull request by LucaMilanese90**
Fixes #376

## Proposed Changes

 - select a compute environment made accessible through the provider of the dataset
 - calculate prices and fees based on the selected environment
 - pass the selected env to the provider on start compute

### CODE REVIEW

**Comment by LucaMilanese90** [comment]
I added the Mumbai testnet and switched to the OPF Provider and Aquarius to be able to test with the new `computeEnvironments` data type, keep in mind that before merging we'll need to either update our own Provider or account for the old (and apparently not supported anymore) data structure.

**Comment by moritzkirstein** [comment]
Will have a more detailled look later, quick finding: the select currently does not switch for me (or at least gives no feedback)

https://github.com/deltaDAO/mvg-portal/assets/28757404/215c62b1-af86-406a-a4db-d5c85079b422

**Comment by LucaMilanese90** [comment]
Yes, you are right, it got messed up when I refactored it to have the field in the JSON file. Should work properly now.

File: content/pages/startComputeDataset.json [hunk 1]
```diff
@@ -3,6 +3,13 @@
     "success": "🎉 Your Compute job started. 🎉",
     "error": "Compute job could not be started.",
     "data": [
+      {
+        "name": "computeEnv",
+        "label": "Select a Compute Environment",
+        "type": "select",
```
Comment (moritzkirstein) [inline content/pages/startComputeDataset.json:9]: note: see comment below on potential new type for this component

File: src/components/Asset/AssetActions/Compute/FormComputeDataset.tsx [hunk 2]
```diff
@@ -86,15 +86,10 @@ export default function FormStartCompute({
...
-    values
-  }: FormikContextType<{
-    algorithm: string
-    computeEnv: string
-  }> = useFormikContext()
+  const { isValid, values, errors }: FormikContextType<ComputeDatasetForm> =
+    useFormikContext()
   const { asset, isAssetNetwork } = useAsset()
-
+  console.log(errors)
```
Comment (moritzkirstein) [inline src/components/Asset/AssetActions/Compute/FormComputeDataset.tsx:92]: console.log can be removed

**Review by moritzkirstein — APPROVED** [review]

**Pull request merged**